### PR TITLE
Created new checks without the redundant TrainTest prefix

### PR DESCRIPTION
**Pull request by nissimofir**
Added deprecation class to help mark classes as deprecated, in addition, deprecated the next classes:
Vision - TrainTestPredictionDrfit, TrainTestLabelDrift
Tabular - TrainTestSamplesMix, TrainTestLabelDrift, TrainTestFeatureDrift, TrainTestPerfromance, TrainTestPredictionDrift

### CODE REVIEW

**Comment by shir22** [comment]
For all drift checks - it makes sense to drop the TrainTest

But for "TrainTestPerformance" -> I think it sounds weird to have a "Performance" check...
specifically since that means we will have "SingleDatasetPerformance" and also "Performance" -> this won't be clear to the innocent user, IMO we should leave it to be TrainTestPerformance

Personally I also think TrainTestSamplesMix is clearer than SamplesMix (making it self explainable that the mix is between two datasets and not within itself or something)

Summing up - I think it should be dropped only for drift checks...

File: deepchecks/utils/deprecation.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        ----------
+        args args
+        kwargs kwargs
+
+        Returns instance of the new class
+        -------
+
+        """
+        self._warn()
+        return self.new_target(*args, **kwargs)
```
Comment (noamzbr) [inline deepchecks/utils/deprecation.py:56]: I'm sorry but I think this is not such a good idea... I'm sure to get the warning when importing the check. Given that, I don't need an additional warning every time I run it or get any attribute...

File: deepchecks/vision/checks/model_evaluation/prediction_drift.py [hunk 2]
```diff
@@ -0,0 +1,308 @@
...
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains Train Test Prediction Drift check."""
```
Comment (noamzbr) [inline deepchecks/vision/checks/model_evaluation/prediction_drift.py:11]:
```suggestion
"""Module contains Prediction Drift check."""
```

File: deepchecks/vision/checks/train_test_validation/label_drift.py [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+# Copyright (C) 2021-2023 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Module contains Train Test label Drift check."""
```
Comment (noamzbr) [inline deepchecks/vision/checks/train_test_validation/label_drift.py:11]:
```suggestion
"""Module contains Label Drift check."""
```

File: docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py [hunk 4]
```diff
@@ -26,7 +26,7 @@
 in which labels are not available for the test dataset, and so a drift in the predictions
 is our only indication that a changed has happened in the data that actually affects model
 predictions. If labels are available, it's also recommended to run the
-:doc:`Label Drift check </checks_gallery/tabular/train_test_validation/plot_train_test_label_drift>`.
+:doc:`Label Drift check </checks_gallery/tabular/train_test_validation/plot_label_drift>`.
```
Comment (noamzbr) [inline docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py:29]: I think that in past deprecation we straight-out deleted the plot examples of the deprecated checks. Alternatively, this can lead to the new doc.

Comment (noamzbr) [reply docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py:29]: @nirhutnik do you remember?

File: docs/source/checks/tabular/model_evaluation/plot_prediction_drift.py [hunk 5]
```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_tabular_prediction_drift:
+
+Train Test Prediction Drift
```
Comment (noamzbr) [inline docs/source/checks/tabular/model_evaluation/plot_prediction_drift.py:5]: Update title and other places where old name may appear

File: docs/source/checks/tabular/train_test_validation/plot_feature_drift.py [hunk 6]
```diff
@@ -0,0 +1,191 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_tabular_feature_drift:
+
+Train Test Feature Drift
```
Comment (noamzbr) [inline docs/source/checks/tabular/train_test_validation/plot_feature_drift.py:5]: Same, update name

File: docs/source/checks/tabular/train_test_validation/plot_label_drift.py [hunk 7]
```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_tabular_label_drift:
+
+Train Test Label Drift
```
Comment (noamzbr) [inline docs/source/checks/tabular/train_test_validation/plot_label_drift.py:5]: Update name

File: docs/source/checks/vision/model_evaluation/plot_prediction_drift.py [hunk 8]
```diff
@@ -0,0 +1,220 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_vision_train_test_prediction_drift:
+
+Train Test Prediction Drift
```
Comment (noamzbr) [inline docs/source/checks/vision/model_evaluation/plot_prediction_drift.py:5]: same

File: docs/source/conf.py [hunk 9]
```diff
@@ -108,7 +108,7 @@
     "user-guide/key_concepts": "../user-guide/general/deepchecks_hierarchy.html",
     "user-guide/when_should_you_use": "../getting-started/when_should_you_use.html",
     "examples/checks/distribution/index": "../../../checks_gallery/tabular/index.html",
-    "examples/checks/distribution/train_test_feature_drift": "../../../checks_gallery/tabular/train_test_validation/plot_train_test_feature_drift.html",
+    "examples/checks/distribution/feature_drift": "../../../checks_gallery/tabular/train_test_validation/plot_feature_drift.html",
```
Comment (noamzbr) [inline docs/source/conf.py:111]: I think these lines have not been in use for a while now, can you ask the last one to modify these lines.

**Review by noamzbr — COMMENTED** [review]
Great work! Main comments not are not on any individual lines:
1. Should change tests to use the new check names
2. Should have deprecation tests that test that the warning is raised when importing the old version of the check. See test_tabular_deprecation_warnings.py
3. Should have something making sure that warnings are raised only at the first import of the deprecated check. See deprecation_warnings.py

File: docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py (hunk 4 above)
Comment (nirhutnik) [reply docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py:29]: Agree, no need to keep the old plot_... etc.

Comment (nissimofir) [reply docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py:29]: Ok, removing all plot_train_test...

File: docs/source/conf.py (hunk 9 above)
Comment (nissimofir) [reply docs/source/conf.py:114]: Waiting for an answer from Itay

File: docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py (hunk 4 above)
Comment (matanper) [reply docs/source/checks/tabular/model_evaluation/plot_train_test_prediction_drift.py:29]: Note that you have `examples_validation.py` in the tests, you need to add the deprecated checks to the `ignored_classes` in order to not check for their plot files.

But if someone will use the deprecated check he will be navigated to a broken link, so not sure what's correct...

**Review by ItayGabbay — APPROVED** [review]

**Review by noamzbr — APPROVED** [review]
Great first PR!
